src/main/java/view/UserPageImpl.java

```diff
@@ -38,15 +67,115 @@ public void showBlindCost() {
         }
     }
 
-    private BlindHorizontal requestData() {
+    private BlindHorizontal requestDataHorizontalBlind() {
         BlindHorizontal blindHorizontal = new BlindHorizontal();
-        System.out.println("Enter the width of the blindHorizontal in mm: ");
-        blindHorizontal.setWidth(scanner.nextInt());
-        System.out.println("Enter the height of the blindHorizontal in mm:");
-        blindHorizontal.setHeight(scanner.nextInt());
-        System.out.println("Enter the color number of the blindHorizontal (201, 202): ");
-        blindHorizontal.setColor(scanner.nextInt());
+        System.out.println("Enter the width of the horizontal blind in mm: ");
+        while (true) {
+            String line = scanner.nextLine();
+            if (!line.matches("^-?\\d+$") || Integer.parseInt(line) < 250 || Integer.parseInt(line) > 2700) {
+                System.out.println("You entered incorrect data, please enter again!");
```

Comment:
need to add all requirements about input number

Reply:
done